tonic/src/transport/channel/tls.rs

```diff
@@ -46,6 +48,32 @@ impl ClientTlsConfig {
         ClientTlsConfig { certs, ..self }
     }
 
+    /// Adds the trust anchor which to verify the server's TLS certificate.
+    pub fn trust_anchor(self, trust_anchor: TrustAnchor<'static>) -> Self {
+        let mut trust_anchors = self.trust_anchors;
+        trust_anchors.push(trust_anchor);
+        ClientTlsConfig {
+            trust_anchors,
+            ..self
+        }
+    }
+
+    /// Adds the multiple trust anchors which to verify the server's TLS certificate.
+    pub fn trust_anchors(
+        self,
+        trust_anchors: impl IntoIterator<Item = TrustAnchor<'static>>,
+    ) -> Self {
+        let trust_anchors = {
+            let mut tmp = self.trust_anchors;
+            tmp.extend(trust_anchors);
+            tmp
+        };
```

Comment:
Seems like a lot of syntactic overhead for the benefit of not having `mut` on `self`? I would just write `self.trust_anchors.extend(trust_anchors)`.

Reply:
Updated.